corehq/apps/locations/resources/v0_6.py

```diff
@@ -35,4 +39,68 @@ def dehydrate(self, bundle):
             bundle.data['parent_location_id'] = ''
         bundle.data['location_type_name'] = bundle.obj.location_type.name
         bundle.data['location_type_code'] = bundle.obj.location_type.code
+        bundle.data['location_data'] = bundle.obj.metadata
+        return bundle
+
+    def obj_create(self, bundle, **kwargs):
+        domain = kwargs['domain']
+        if 'name' not in bundle.data:
+            raise BadRequest("'name' is a required field.")
+        if SQLLocation.objects.filter(domain=domain, site_code=bundle.data['site_code']).exists():
+            raise BadRequest("Location on domain with site code already exists.")
+        bundle.obj = SQLLocation(domain=domain)
+        self._update(bundle)
         return bundle
+
+    def obj_update(self, bundle, **kwargs):
+        try:
+            bundle.obj = SQLLocation.objects.get(location_id=kwargs['location_id'])
+        except SQLLocation.DoesNotExist:
+            raise BadRequest("Could not find location with given ID.")
+        if bundle.obj.domain != kwargs.pop('domain'):
+            raise NotFound('Location could not be found.')
+        self._update(bundle)
+        return bundle
+
+    def _update(self, bundle):
+        data = bundle.data
+        # Invalid fields that might be common
+        if data.pop('location_type_name', False):
+            raise BadRequest('Location type name is not editable.')
+        if data.pop('last_modified', False):
+            raise BadRequest('\"last_modified\" field is not editable.')
+
+        easy_field_names = ['name', 'site_code', 'latitude', 'longitude']
```

Comment:
`site_code` can't contain special characters, check out `valid_location_site_code`.

Reply:
it is actually enforced on the model, just down in the `Meta`:
https://github.com/dimagi/commcare-hq/blob/b6984b8a4758eb740945855b19b677c74d420237/corehq/apps/locations/models.py#L582-L582
I agree on the expansion of scope, though I think it's valuable.  (We offlined, for anyone reading this)